static/js/stream_edit.js

```diff
@@ -652,11 +652,11 @@ exports.initialize = function () {
         if (!sub.subscribed) {
             regular_sub_settings.addClass("in");
             exports.open_edit_panel_for_row(stream_row);
+            setup_subscriptions_stream_hash(sub);
```

Comment:
To reproduce the issue:
 - Open stream settings of stream "A"
 - Unsubscribe user from stream "B" by clicking on
   subscription-checkmark in stream lists.
 - The URL is changed to unsubscribed stream
 - Stream settings of stream "A" is still open

This issue further create error when `active_stream` function returns stream id present in URL, but actually that stream is not active in UI, so it throws ui-elements-not-found errors.

Reply:
Can you explain why this is the correct fix for that issue?  I don't understand why moving this into the if block helps.

Reply:
I have update the function itself to call `setup_subscriptions_stream_hash` inside `open_edit_panel_for_row`.
The reason that I put that in `if` block was that `setup_subscriptions_stream_hash` function should be called whenever `open_edit_panel_for_row` function is called and vice versa. This way the UI and URL stays consistent. If these functions are not called simultaneously, then there will be case where URL is of stream "A" and UI is of stream "B" or empty. This will return false result in `exports.active_stream()` function in `static/js/subs.js`